OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -521,13 +521,21 @@ public void UnReserve(bool takeOff = false)
 			if (reservation == null)
 				return;
 
+			// Move to the host's rally point if it has one
+			var rp = ReservedActor != null ? ReservedActor.TraitOrDefault<RallyPoint>() : null;
+
 			reservation.Dispose();
 			reservation = null;
 			ReservedActor = null;
 			MayYieldReservation = false;
 
 			if (takeOff && self.World.Map.DistanceAboveTerrain(CenterPosition).Length <= LandAltitude.Length)
-				self.QueueActivity(new TakeOff(self));
+			{
+				if (rp != null)
+					self.QueueActivity(new TakeOff(self, Target.FromCell(self.World, rp.Location)));
```

Comment:
Why call `TakeOff` here but `move.MoveTo` in `Resupply`?
If you call  `move.MoveTo` here as well (optionally wrapped in `AttackMoveActivity` if that is indeed the desired behaviour) then you can get rid of the target handling in `TakeOff` completely.

Reply:
That was originally my plan, but then I realized that `TakeOff` with a target is not equivalent to `move.MoveTo` when there are activities queued afterwards. Trying to untangle and rewrite the different cases vastly increases the scope and regression potential of the PR.

The `move.MoveTo` in `Resupply` is copied directly from the `Mobile` branch, with the goal of having similar behaviour for all units.